src/domain/resource/utils.js

```diff
@@ -87,3 +95,112 @@ export const isFree = (resource) => {
+    opens: get(openingHours, `[${index}].opens`),
+  };
+};
+
+const reservingIsRestricted = (resource, date) => {
+  if (!date) {
+    return false;
+  }
+
+  const isAdmin = get(resource, 'user_permissions.is_admin', false);
+  const isLimited = resource.reservable_before && moment(resource.reservable_before).isBefore(moment(date), 'day');
+
+  return isLimited && !isAdmin;
+};
+
+export const getOpenReservations = (resource) => {
+  return filter(resource.reservations, (reservation) => {
+    return reservation.state !== 'cancelled' && reservation.state !== 'denied';
+  });
+};
+
+export const getAvailabilityDataForNow = (resource, date) => {
+  const openingHours = getOpeningHours(resource, date);
+  const reservations = getOpenReservations(resource);
+
+  if (!openingHours || !openingHours.closes || !openingHours.opens) {
+    return { status: 'closed', bsStyle: 'danger' };
+  }
+
+  const nowMoment = moment();
+  const opensMoment = moment(openingHours.opens);
+  const closesMoment = moment(openingHours.closes);
+  const beginMoment = nowMoment > opensMoment ? nowMoment : opensMoment;
+
+  if (nowMoment > closesMoment) {
+    return { status: 'closed', bsStyle: 'danger' };
+  }
+
+  const currentReservation = reservationUtils.getCurrentReservation(reservations);
+  if (currentReservation || nowMoment < opensMoment) {
+    const nextAvailableTime = reservationUtils.getNextAvailableTime(reservations, beginMoment);
+    if (nextAvailableTime < closesMoment) {
+      return {
+        status: 'availableAt',
+        bsStyle: 'danger',
+        values: { time: nextAvailableTime.format(constants.TIME_FORMAT) },
+      };
+    }
+    return { status: 'reserved', bsStyle: 'danger' };
+  }
+
+  return { status: 'available', bsStyle: 'success' };
+};
+
+export const getAvailabilityDataForWholeDay = (resource, date) => {
+  const openingHours = getOpeningHours(resource, date);
+
+  if (!openingHours || !openingHours.closes || !openingHours.opens) {
+    return { status: 'closed', bsStyle: 'danger' };
+  }
+
+  if (reservingIsRestricted(resource, date)) {
+    return { status: 'reservingRestricted', bsStyle: 'danger' };
+  }
+
+  const reservations = getOpenReservations(resource);
+
+  const opensMoment = moment(openingHours.opens);
+  const closesMoment = moment(openingHours.closes);
+
+  let total = closesMoment - opensMoment;
+  forEach(reservations, (reservation) => {
+    const resBeginMoment = moment(reservation.begin);
+    if (!resBeginMoment.isSame(opensMoment, 'd')) {
+      return;
+    }
+
+    const resEndMoment = moment(reservation.end);
+    total -= resEndMoment;
+    total += resBeginMoment;
+  });
+
+  const asHours = moment.duration(total).asHours();
+  const rounded = Math.ceil(asHours * 2) / 2;
+
+  if (rounded === 0) {
+    return { status: 'reserved', bsStyle: 'danger' };
+  }
+
+  return {
+    status: 'availableTime',
+    bsStyle: 'success',
+    values: { hours: rounded },
+  };
+};
```

Comment:
And unit tests for these as well :)

Reply:
I didn't bother to start building tests for these, because the old code didn't have tests either.